addon/-private/service-injections.js

```diff
@@ -0,0 +1,15 @@
+const SERVICE_INjECTIONS = new WeakMap();
+
+export function setServiceInjections(fn, services = []) {
+  SERVICE_INjECTIONS.set(fn, services);
+}
+
+export function getServiceInjections(fn, owner) {
```

Comment:
In theory, you could cache the final looked up services instead of looking them up for each modifier invocation. It shouldn't complicate the implementation too much, but it would probably have at least _some_ impact to render times if a single modifier would be used more than once...

Reply:
That might look like:

```js
// addon/-private/functional-manager.js

// ...snip...
export default class FunctionalModifierManager {
  constructor(owner) {
    this.owner = owner;
    this.serviceCache = new WeakMap();
  }

  getServicesFor(fn) {
    let services = this.serviceCache.get(fn);
	if (services === undefined) {
   	  services = getServiceInjections(fn, this.owner);
      this.serviceCache.set(fn, services);
    }

    return services;
  }
  // ...snip...
}
```